lib/CGI.pm

```diff
@@ -622,16 +622,8 @@ sub init {
 	  last METHOD;
       }
 
-      # If method is GET, HEAD or DELETE, fetch the query from
-      # the environment.
-      if ($is_xforms || $meth=~/^(GET|HEAD|DELETE)$/) {
-          $query_string = $self->_get_query_string_from_env;
-         $self->param($meth . 'DATA', $self->param('XForms:Model'))
-             if $is_xforms;
-	      last METHOD;
-      }
-
-      if ($meth eq 'POST' || $meth eq 'PUT') {
+      # First, let's try to get data from request body
+      if ($meth=~/^(GET|HEAD|DELETE|POST|PUT)$/) {
```

Comment:
You've added GET and HEAD here, which is not only meaningless but goes against both the IETF RFC and common convention. See [https://tools.ietf.org/html/rfc2616#section-9.1.1](https://tools.ietf.org/html/rfc2616#section-9.1.1) and [https://groups.yahoo.com/neo/groups/rest-discuss/conversations/messages/9962](https://groups.yahoo.com/neo/groups/rest-discuss/conversations/messages/9962). I think it's a bad idea to allow this even if convetion/RFCs are explicit because it's encouraging bad API design.